src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp

```diff
@@ -44,8 +46,29 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
      */
     std::shared_ptr<TScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
     {
+        nlohmann::json response;
+
+        try
+        {
+            // Execute query
+            TSocketDBWrapper::instance().query(
+                WazuhDBQueryBuilder::builder()
+                    .globalGetCommand(std::string("agent-info ") + data->agentId().data())
+                    .build(),
+                response);
+        }
+        // LCOV_EXCL_START
+        catch (const std::exception& e)
+        {
+            logDebug2(WM_VULNSCAN_LOGTAG, "Error executing query to fetch global agent data. Reason: %s.", e.what());
+            throw WdbDataException("Error executing query to fetch global agent data", data->agentId().data());
+        }
+        // LCOV_EXCL_STOP
+        // Agent in the first element of the array.
+        const auto& agent = response.back();
         data->m_agents.push_back(
-            {data->agentId().data(), data->agentName().data(), data->agentVersion().data(), data->agentIp().data()});
+            {data->agentId().data(), agent.at("name"), Utils::leftTrim(agent.at("version"), "Wazuh "), agent.at("ip")});
+
```

Comment:
The response will be an empty array if no information is retrieved from an agent (eg, DB doesn't exist).

Accessing an empty array with back is undefined behavior ([see example](https://json.nlohmann.me/api/basic_json/back/#examples))

We can handle this by verifying the response is not empty